### PR TITLE
fix(phone/store): reset character specific data properly for rehydration 

### PR DESCRIPTION
**Pull request by TasoOneAsia**
**Pull Request Description**

This should eliminates the selector caching, which resolves data from the Recoil cache rather than making another request. This can most likely be removed upon merge of the new `useRecoilCallback` API which passes a refresh property to the useRecoilCallback cb.

See the following issue for further context. See [facebookexperimental/recoil](https://github.com/facebookexperimental/Recoil/issues/972) for further context on RecoilRoot reset limitations.

*Thankfully it appears that reset/cache-control API functionality is being further addressed in upcoming releases.*

**Pull Request Checklist**:
* [x] Have you followed the guidelines in our Contributing document and Code of Conduct?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/project-error/npwd/pulls) for the same update/change?
* [x] Have you built and tested NPWD in-game after the relevant change?

### CODE REVIEW

**Comment by RockySouthpaw** [comment]
Will close https://github.com/project-error/npwd/issues/476 as well

**Comment by PERPGamer** [comment]
I tested this by compiling it and loading it up on my server. For some reason Twitter, for example, data doesn't load
With pull:
image

Without pull:
image

File: phone/src/lib/RecoilRootManager.tsx [hunk 1]
```diff
@@ -1,20 +1,43 @@
...
+  const [isNotChar, setIsNotChar] = useState(false);
 
-  useNuiEvent('PHONE', PhoneEvents.UNLOAD_CHARACTER, () => {
-    setCharState((charState) => charState + 1);
+  useNuiEvent<boolean>('PHONE', PhoneEvents.UNLOAD_CHARACTER, (val) => {
+    setIsNotChar(val);
   });
 
-  return <RecoilRoot key={charState}>{children}</RecoilRoot>;
+  return !isNotChar ? (
```
Comment (itschip) [inline phone/src/lib/RecoilRootManager.tsx:37]: Simplify with `&&`

**Comment by PERPGamer** [comment]
I am properly getting annoying now. With itschip's change, it seems to have fixed the twitter issue but whenever you change characters, your phone will not load back.

**Comment by RockySouthpaw** [comment]
> I am properly getting annoying now. With itschip's change, it seems to have fixed the twitter issue but whenever you change characters, your phone will not load back.

idk why youd get annoyed this is a **draft** pr. In the future, please be aware **draft** pr's are not meant for prod and are work in progressess.

**Comment by RockySouthpaw** [comment]
As stated above, this seems to break when swapping characters.

Reproduction:
1. Join server on char 1
2. Open phone and view twitter profile
3. Use a character change command that sends the unload player
4. Select any character (Even the one you just had)
5. Open phone

phone won't open, no errors.

**Comment by PERPGamer** [comment]
> 

I wasn't getting mad :( I was saying I was probably being annoying with responding but I said properly, I apologize

**Comment by TasoOneAsia** [comment]
Although I really don't want to write my own atom manager, if this approach doesn't fix the issue, I may have to. I still wonder why this isn't working though.

**Comment by DocBurN** [comment]
I just want to vote on the importance of the character reload feature and my love for the phone. I do switch character a lot as admin .. thank you for working on this guys..

**Comment by dolutattoo** [comment]
> As stated above, this seems to break when swapping characters.
> 
> Reproduction:
> 
> 1. Join server on char 1
> 2. Open phone and view twitter profile
> 3. Use a character change command that sends the unload player
> 4. Select any character (Even the one you just had)
> 5. Open phone
> 
> phone won't open, no errors.
> 
> Here is unload code:
> 
> ```lua
> if Config.useNPWD and GetResourceState('npwd') == 'started' then
>             exports.npwd:unloadPlayer(playerId)
>         end
> ```

I can reproduce exactly the same. Don't even need to open the phone, once you use the export, the phone won't appear anymore.

**Comment by RockySouthpaw** [comment]
> > As stated above, this seems to break when swapping characters.
> > Reproduction:
> > 
> > 1. Join server on char 1
> > 2. Open phone and view twitter profile
> > 3. Use a character change command that sends the unload player
> > 4. Select any character (Even the one you just had)
> > 5. Open phone
> >    ...
> 
> I can reproduce exactly the same. Don't even need to open the phone, once you use the export, the phone won't appear anymore.

This doesn’t really help as we’re already aware this PR is broken. When @TasoOneAsia is actually going to finish it idk

**Comment by itschip** [comment]
@AvarianKnight I believe we can close this, right?

**Pull request closed without merging**